src/theme.jl

```diff
@@ -393,3 +391,82 @@ end
+    line_width::Measure=0.3mm,
+    line_style::Union{Symbol,Vector}=:solid,
+    panel_fill::ColorOrNothing=nothing,
+    panel_stroke::ColorOrNothing=nothing,
+    panel_opacity::Float64=1.0,
+    background_color::ColorOrNothing=nothing,
+    plot_padding::(Vector{<:Measure})=[5mm],
+    grid_color::ColorOrNothing=colorant"#D0D0E0",
+    grid_line_style::Union{Symbol,Vector}=[0.5mm, 0.5mm],
+    grid_color_focused::ColorOrNothing=colorant"#A0A0A0",
+    grid_line_width::Measure=0.2mm,
+    minor_label_font::AbstractString=label_font_desc,
+    minor_label_font_size::Measure=8pt,
+    minor_label_color::ColorOrNothing=colorant"#6c606b",
+    major_label_font::AbstractString=title_font_desc,
+    major_label_font_size::Measure=11pt,
+    major_label_color::ColorOrNothing=colorant"#564a55",
+    point_label_font::AbstractString=label_font_desc,
+    point_label_font_size::Measure=8pt,
+    point_label_color::ColorOrNothing=colorant"#4c404b",
+    key_title_font::AbstractString=title_font_desc,
+    key_title_font_size::Measure=11pt,
+    key_title_color::ColorOrNothing=colorant"#362a35",
+    key_label_font::AbstractString=title_font_desc,
+    key_label_font_size::Measure=8pt,
+    key_label_color::ColorOrNothing=colorant"#4c404b",
+    key_color_gradations::Int=40,
+    bar_spacing::Measure=-0.05mm,
+    boxplot_spacing::Measure=1mm,
+    errorbar_cap_length::Measure=3mm,
+    stroke_color::Function=default_stroke_color,
+    highlight_width::Measure=0.3mm,
+    discrete_highlight_color::Function=default_discrete_highlight_color,
+    continuous_highlight_color::Function=default_continuous_highlight_color,
+    lowlight_color::Function=default_lowlight_color,
+    lowlight_opacity::Float64=NaN,
+    middle_color::Function=default_middle_color,
+    middle_width::Measure=0.6mm,
+    guide_title_position::Symbol=:left,
+    colorkey_swatch_shape::Symbol=:square,
+    key_swatch_shape::Function=Shape.square,
+    key_swatch_color::ColorOrNothing=nothing,  
+    key_position::Symbol=:right,
+    bar_highlight::Union{(Void), Function, Color}=nothing,
+    rug_size::Measure=2.0mm,
+    label_placement_iterations::Int=1000,
+    label_out_of_bounds_penalty::Float64=10.0,
+    label_hidden_penalty::Float64=0.5,
+    label_visibility_flip_pr::Float64=0.2,
+    label_padding::Measure=1mm,
+    key_max_columns::Int=4,
+    discrete_color_scale::Scale.DiscreteColorScale=Scale.color_discrete(),
+    continuous_color_scale::Scale.ContinuousColorScale=Scale.color_continuous()
+)
+    
+    isfinite(lowlight_opacity) && Base.depwarn("The keyword argument `lowlight_opacity` has been deprecated, and never worked anyway!", :Theme)
+    
+return Theme(default_color, point_size, point_size_min, point_size_max, point_shapes, line_width, line_style, panel_fill, panel_stroke, panel_opacity, background_color, 
```

Comment:
would `Theme(default_color==nothing ? nothing : parse_colorant(default_color), ....` work?  this is how it's done for the non-deprecated constructor https://github.com/GiovineItalia/Gadfly.jl/blob/master/src/varset.jl#L47

Reply:
But that's no different then making a new `Theme` constructor, in which case the varset constructor for `Theme` is obsolete.  If the `Theme` constructor is coded explicitly then yes it easy to do deprecations. Removing the varset `Theme` constructor and making a new constructor should really be done as  a separate PR.

Reply:
i think all we need here is

```
function Theme(; default_color, point_size, ..., lowlight_opacity, ...)
    Base.depwarn(....)
    Theme(
        default_color==nothing ? nothing : parse_colorant(default_color),
        point_size,
        ...
        #skip lowlight_opacity,
        ...)
end
```

no need to specify the types, as that will be handled by the varset definition of the Theme object.